GooglePlayInstant/Editor/QuickDeploy/AccessTokenGetter.cs

```diff
@@ -0,0 +1,191 @@
+
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+
+namespace GooglePlayInstant.Editor.QuickDeploy
+{
+    /// <summary>
+    /// Provides methods for using the OAuth2 flow to get user authorization and to retrieve an access token that is
+    /// used send HTTP requests to Google Cloud Platform APIs.
+    /// </summary>
+    public static class AccessTokenGetter
+    {
+        private const string OAuth2CodeGrantType = "authorization_code";
+
+        // Full control scope is required, since the application needs to read, write and change access
+        // permissions of buckets and objects.
+        private const string CloudStorageFullControlScope = "https://www.googleapis.com/auth/devstorage.full_control";
+        private static KeyValuePair<string, string>? _authorizationResponse;
+
+        private static Action<KeyValuePair<string, string>> _onOAuthResponseReceived;
+
+        /// <summary>
+        /// Get Access token to use for API calls if available. Returns null if access token is not available.
+        /// </summary>
+        public static GcpAccessToken AccessToken { get; private set; }
+
+        /// <summary>
+        /// Check whether a new authorization code has been received and execute scheduled tasks accordingly.
+        /// </summary>
+        public static void Update()
+        {
+            // Handle scheduled tasks for when authorization code is received.
+            if (_authorizationResponse.HasValue && _onOAuthResponseReceived != null)
+            {
+                _onOAuthResponseReceived(_authorizationResponse.Value);
+                _authorizationResponse = null;
+                _onOAuthResponseReceived = null;
+            }
+        }
+
+        /// <summary>
+        /// Get new access token if access token is expired or is not available, and execute the action when the access
+        /// token is available.
+        /// </summary>
+        /// <param name="postTokenAction">Action to be executed when valid access token is avalable.</param>
+        public static void UpdateAccessToken(Action postTokenAction)
+        {
+            // TODO: Implement reuse of refresh token to get a new access token when the current token is expired.
+            if (AccessToken == null)
+            {
+                GetAuthCode(code => RequestAndStoreAccessToken(code, postTokenAction));
+            }
+            else
+            {
+                postTokenAction();
+            }
+        }
+
+        /// <summary>
+        /// Instantiate the OAuth2 flow to retrieve authorization code for google cloud storage, and schedule
+        /// invocation of the code handler on the received authorization code once it is available or throw an exception
+        /// once there is a failure to get the authorization code.
+        /// </summary>
+        /// <param name="authorizationCodeHandler"> An action to invoke on the authorization code instance when it is
+        /// available.></param>
```

Comment:
Nits: When you fixed this up you still need ti delete the ">" after "available." - also delete space before "An action".